File: cpp/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> seen;
        for (int i = 0; i < nums.size(); i++) {
            if (seen.find(target - nums[i]) != seen.end()) {
                return {i, seen[target - nums[i]]};
            }
            seen[nums[i]] = i;
        }
        return {};
    }
}; 

File: csharp/0001-two-sum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> seen = new Dictionary<int, int>();
        for (int i = 0; i < nums.length; i++) {
            if (seen.ContainsKey(target - nums[i])) {
                return new int[] {i, seen[target - nums[i]]};
            }
            seen[nums[i]] = i;
        }
        return new int[] {};
    }
} 